src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3109,6 +3158,7 @@ internal Buffer(IEnumerable<TElement> source)
                     count++;
                 }
             }
+
             this.items = items;
```

Comment:
I was hesitant to change it for that reason.  Is it ok to make it Array.Empty<TElement>()? (Regardless, if we want to change it I'd like us to do it as part of a separate PR / commit.)